### ISSUE TITLE
`-Wunused-const-variable` incorrectly warns for exported lambda

### ISSUE THREAD

**Issue by davidstone** [llvm/llvm-project#85122]
Compiling the following translation unit with `-Wunused-const-variable`:

```cpp
export module a;

export constexpr auto a = []{};
```

causes clang to report

```console
<source>:3:23: warning: unused variable 'a' [-Wunused-const-variable]
    3 | export constexpr auto a = []{};
      |                       ^
1 warning generated.
Compiler returned: 0
```

See it live: https://godbolt.org/z/555Y3q9fe

Anything exported from a module should never be considered unused.

**Comment by shafik** [comment]
CC @ChuanqiXu9
